### PR TITLE
REM-291

### PR DESCRIPTION
**Pull request by imtithalhalalo**
1- Add new record check transaction log it should show records
2- edit any record click “clear” nothing happen

### CODE REVIEW

File: src/pages/pos-users/forms/PosUsersForm.js [hunk 1]
```diff
@@ -13,19 +13,21 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
     endpointId: PointofSaleRepository.PosUsers.qry
   })
 
-  const editMode = !!userId
+
 
   const { formik } = useForm({
     initialValues: {
-      recordId: null,
+      recordId: recordId || null,
```
Comment (omarZrayka23) [inline src/pages/pos-users/forms/PosUsersForm.js:30]: remove || null

File: src/pages/pos-users/forms/PosUsersForm.js [hunk 2]
```diff
@@ -34,32 +36,35 @@ export default function PosUsersForm({ labels, maxAccess, userId, window }) {
...
       try {
-        if (userId) {
+        if (record && record.userId && recordId) {
           const res = await getRequest({
             extension: PointofSaleRepository.PosUsers.get,
-            parameters: `_userId=${userId}`
+            parameters: `_userId=${record.userId}`
           })
-          formik.setValues({ ...res.record, recordId: res?.record?.userId })
+          formik.setValues({ ...res.record, recordId: res.record.userId })
```
Comment (omarZrayka23) [inline src/pages/pos-users/forms/PosUsersForm.js:67]: use setFieldvalue instead setValues

Comment (omarZrayka23) [inline src/pages/pos-users/forms/PosUsersForm.js:67]: formik.setFieldValue('spId', newValue ? newValue.recordId : '') instead 
formik.setFieldValue('spId', newValue.?recordId )

**Comment by omarZrayka23** [comment]
[
Screenshot 2024-08-29 at 2 05 10 PM
](url).   when i  click submit in editMode after  i  clears pos value  , pos error required  is not occur

File: src/pages/pos-users/forms/PosUsersForm.js (hunk 2 above)
Comment (imtithalhalalo) [reply src/pages/pos-users/forms/PosUsersForm.js:67]: here we are setting the res.record + the recordId

File: src/pages/pos-users/forms/PosUsersForm.js [hunk 3]
```diff
@@ -102,7 +107,7 @@ export default function PosUsersForm({ labels, maxAccess, userId, window }) {
                 required
                 maxAccess={maxAccess}
                 onChange={(event, newValue) => {
-                  formik.setFieldValue('posId', newValue?.recordId)
+                  formik.setFieldValue('posId', newValue ? newValue?.recordId : '')
```
Comment (omarZrayka23) [inline src/pages/pos-users/forms/PosUsersForm.js:110]: formik.setFieldValue('posId', newValue?.recordId )

**Review by omarZrayka23 — DISMISSED** [review]

**Pull request merged**